src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,248 @@
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  // Extracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // This is very dangerous, Let's add a check tommorrow.
+  std::string column_delimiter;
+  if (CLI::HasParam("column_delimiter"))
+  {
+    column_delimiter = CLI::GetParam<std::string>("column_delimiter");
+    // Allow only 3 delimeter.
+    RequireParamValue<std::string>("column_delimiter", [](std::string del)
+        { return del == "\t" || del == "," || del == " "; }, true,
+        "Delimiter should be either \\t (tab) or , (comma) or ' ' (space) ");
+  }
+  else
+  {
+      if (data::Extension(filename) == "csv")
+      {
+        column_delimiter = ",";
+        Log::Warn << "Found csv Extension, taking , as column_delimiter. \n";
+      }
+      else if (data::Extension(filename) == "tsv" ||
+          data::Extension(filename) == "txt")
+      {
+        column_delimiter = "\t";
+        Log::Warn << "Found tsv or txt Extension, taking \\t as"
+        "column_delimiter. \n";
+      }
+      else
+      {
+        Log::Warn << "column_delimiter not specified, taking default value \n";
+      }
+  }
+  // Handling Dimension vector
+  std::vector<std::string> temp_dimension = CLI::GetParam<std::vector<
+    std::string> >("dimension");
+  std::vector<size_t>dimension;
+  int columnstartindex, columnendindex;
+  size_t found;
+  for (size_t i = 0; i < temp_dimension.size(); i++)
+  {
+    found = temp_dimension[i].find('-');
+    if ( found != string::npos)
+    {
+      // Has a range include, something of type a-b.
+      columnstartindex = std::stoi(temp_dimension[i].substr(0, found));
+      columnendindex = std::stoi(temp_dimension[i].substr(found+1,
+          temp_dimension[i].length()));
+      for (int i = columnstartindex; i <= columnendindex; i++)
+      {
+        dimension.push_back(i);
+      }
+    }
+    else
+    {
+      dimension.push_back(std::stoi(temp_dimension[i]));
+    }
+  }
+  std::cout<<"\n";
+  // Sorting neccessary
+  std::sort(dimension.begin(), dimension.end());
+  // Extracting the Contents of file
+  // File pointer
+  ifstream fin;
+  // Open an existing file
+  fin.open(filename);
+  if (!fin.is_open())
+  {
+    throw std::runtime_error("Unable to open input file");
+  }
+  std::string line, word;
+  std::vector<std::vector<std::string>> dataset;
+  boost::string_view token, copy;
+  stringstream streamLine;
+  while (std::getline(fin, line))
+  {
+    streamLine.str(line);
+    dataset.push_back(std::vector<std::string>());
+    // delimeter[0] becase the standard function accepts char as input.
+    while (std::getline(streamLine, word, column_delimiter[0]))
+    {
+      dataset[dataset.size()-1].push_back(word);
+    }
+    streamLine.clear();
+  }
+  size_t col = 0;
+  // Preparing the input dataset on which string manipulation has to be done.
+  std::vector<std::vector<std::string>> input(dimension.size());
```

Comment:
It's hard to understand the difference between `dataset` and `input`. Could you find a bit more meaningful names?